mesonbuild/interpreterbase.py

```diff
@@ -926,6 +928,18 @@ def evaluate_ternary(self, node: mparser.TernaryNode) -> TYPE_var:
         else:
             return self.evaluate_statement(node.falseblock)
 
+    def evaluate_fstring(self, node: mparser.FormatStringNode) -> TYPE_var:
+        assert(isinstance(node, mparser.FormatStringNode))
+
+        def replace(match: T.Match[str]) -> str:
+            var = str(match.group(1))
+            try:
+                return str(self.variables[var])
```

Comment:
We should check the type of variable, probable limit it to str, into and bool.

Reply:
Alright, I'll add type checks.